docs/tutorials/chatbot.md

```diff
@@ -0,0 +1,577 @@
+        self._docs.extend(docs)
+
+    @requests(on='/search')
+    def search(self, docs: 'DocumentArray', **kwargs):
+        """Append best matches to each document in docs
+
+        :param docs: documents that are searched
+        :param kwargs: other keyword arguments
+        """
+        docs.match(self._docs, metric='cosine', normalization=(1, 0), limit=1)
+```
+
+`MyIndexer` exposes 2 endpoints: `index` and `search`. To perform indexing, we use 
+[DocumentArrayMemmap](https://docs.jina.ai/api/jina.types.arrays.memmap/#jina.types.arrays.memmap.DocumentArrayMemmap)` 
+which is a Jina data type. Indexing is a simple as adding the documents to the `DocumentArrayMemmap`.
+
+```{admonition} See Also
+:class: seealso
+Learn more about {ref}`DocumentArrayMemmap<documentarraymemmap-api>`.
+```
+To perform the search operation, we use the method `match` which will return the top match for the query documents 
+using the cosine similarity.
+
+```{admonition} See Also
+:class: seealso
+`.match` is a method of both `DocumentArray` and `DocumentArrayMemmap`. Learn more about it {ref}`in this section<match-documentarray>`.
+```
+
+To import the executors, just add this to the imports:
+
+``` python
+from my_executors import MyTransformer, MyIndexer
+```
+
+And remove the dummy executors we made. Your `app.py` should now look like this:
+
+``` python
+import os
+import webbrowser
+from pathlib import Path
+from jina import Flow, Executor
+from jina.logging.predefined import default_logger
+from jina.types.document.generators import from_csv
+from my_executors import MyTransformer, MyIndexer
+
+
+def tutorial(port_expose):
+    flow = (
+        Flow(cors=True)
+            .add(name='MyTransformer', uses=MyTransformer)
+            .add(name='MyIndexer', uses=MyIndexer)
+    )
+    with flow, open('dataset.csv') as fp:
+        flow.index(from_csv(fp, field_resolver={'question': 'text'}))
+        
+        # switch to REST gateway at runtime
+        flow.protocol = 'http'
+        flow.port_expose = port_expose
+        url_html_path = 'file://' + os.path.abspath(
+            os.path.join(
+                os.path.dirname(os.path.realpath(__file__)), 'static/index.html'
+            )
+        )
+        try:
+            webbrowser.open(url_html_path, new=2)
+        except:
+            pass  # intentional pass, browser support isn't cross-platform
+        finally:
+            default_logger.success(
+                f'You should see a demo page opened in your browser, '
+                f'if not, you may open {url_html_path} manually'
+            )
+        flow.block()
+if __name__ == '__main__':
+    tutorial(8080)
+```
+
+And your directory should be:
+
+    .
+    └── project                    
+        ├── app.py          
+        ├── my_executors.py         
+        ├── static/         
+        ├── our_flow.svg #This will be here if you used the .plot() function       
+        └── dataset.csv
```

Comment:
```suggestion
    tutorial                    
    ├── app.py          
    ├── my_executors.py         
    ├── static/         
    ├── our_flow.svg #This will be here if you used the .plot() function       
    └── dataset.csv
```

Reply:
done